tests/components/unifi/test_device_tracker.py

```diff
@@ -303,6 +311,22 @@ async def test_tracked_devices(hass):
     device_1 = hass.states.get("device_tracker.device_1")
     assert device_1.state == STATE_UNAVAILABLE
 
+    # Update device registry when device is upgraded
+    device_2_copy = copy(DEVICE_2)
+    device_2_copy["version"] = EVENT_DEVICE_2_UPGRADED["version_to"]
+    message = {"meta": {"message": MESSAGE_DEVICE}, "data": [device_2_copy]}
+    controller.api.message_handler(message)
+    event = {"meta": {"message": MESSAGE_EVENT}, "data": [EVENT_DEVICE_2_UPGRADED]}
+    controller.api.message_handler(event)
+    await hass.async_block_till_done()
+
+    # Verify device registry has been updated
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = entity_registry.async_get("device_tracker.device_2")
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+    assert device.sw_version == EVENT_DEVICE_2_UPGRADED["version_to"]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)